mqtt/mqtt-broker/tests/translation.rs

```diff
@@ -0,0 +1,264 @@
+use matches::assert_matches;
+use mqtt3::{
+    proto::{ClientId, QoS},
+    ReceivedPublication,
+};
+
+use common::{TestClient, TestClientBuilder};
+use mqtt_broker::{AuthId, BrokerBuilder};
+
+mod common;
+
+// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#retrieving-a-device-twins-properties
+#[tokio::test]
+async fn translation_twin_retrieve() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let mut server_handle = common::start_server(broker);
+
+    let mut edge_hub_core = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("edge_hub_core".into()))
+        .build();
+    let mut device_1 = TestClientBuilder::new(server_handle.address())
+        .client_id(ClientId::IdWithCleanSession("device_1".into()))
```

Comment:
I think we should start playing with more complex device names. Not necessary in this PR, but we need to test real prod-line device ids. Also, maybe add proptests for ids.

Reply:
I just tried this allowed device name and it breaks the test. I think we need a work item and a separate PR to add proptests and fix all corner cases.
```
fasdfsfd67iasdifu2341_(floor1)-=+*asfd!@#$%',.?das_1
```